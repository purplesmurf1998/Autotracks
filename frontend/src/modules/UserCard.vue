<template>
  <CCard>
    <CCardBody class="p-3">
      <h6>{{ user.first_name }} {{ user.last_name }}</h6>
      <a class="remove-link" @click="removeUser(index)" href="#"
        >Delete staff account</a
      >
      <p class="text-muted mb-0">Role: {{ user.role }}</p>
      <p class="text-muted mb-0">Email: {{ user.email }}</p>
      <CRow class="d-flex justify-content-start mt-3 px-3">
        <CButton color="primary" class="mr-2" @click="setEditUser"
          >Edit</CButton
        >
      </CRow>
    </CCardBody>
  </CCard>
</template>

<script>
export default {
  props: ["user"],
  methods: {
    removeUser() {
      console.log("remove user" + user._id);
    },
    setEditUser() {},
  },
};
</script>

<style scoped>
.remove-link {
  color: red;
}
</style>