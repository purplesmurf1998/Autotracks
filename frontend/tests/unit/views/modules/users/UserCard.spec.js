import {createLocalVue, shallowMount} from '@vue/test-utils'
import CoreuiVue from '@coreui/vue-pro'
import Vuex from 'vuex'

import UserCard from './../../../../../src/modules/users/UserCard.vue'

const localVue = createLocalVue()
localVue.use(CoreuiVue)
localVue.use(Vuex)

const store = new Vuex.Store({
    state: {
      sidebarShow: 'responsive',
      sidebarMinimize: false,
      asideShow: false,
      darkMode: false
    }
  })

  describe('UserCard.vue', () => {
    const wrapper = shallowMount(UserCard, { 
      store, 
      localVue
    })
    it('has a name', () => {
      expect(UserCard.name).toBe('UserCard')
    })
    it('should render correct content', () => {

    })
    test('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })