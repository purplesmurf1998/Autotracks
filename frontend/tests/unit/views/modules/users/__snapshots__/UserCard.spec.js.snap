// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard.vue renders correctly 1`] = `
<ccard-stub>
  <ccardbody-stub
    class="p-3"
  >
    <view-user-stub />
     
    <!---->
  </ccardbody-stub>
</ccard-stub>
`;
